windows/src/developer/kmcmpdll/Compiler.cpp

```diff
@@ -2910,440 +2920,450 @@ BOOL CheckStoreUsage(PFILE_KEYBOARD fk, int storeIndex, BOOL fIsStore, BOOL fIsO
+        continue;
       }
-			if( *p == currentQuotes ) currentQuotes = 0;
-			continue;
-		}
-		if( InComment ) {
-			if( *p == L'\n' ) break;
-			*p = L' ';
-			continue;
-		}
-		if( *p == L'\\' ) {
-			LineCarry = TRUE;
-			*p = L' ';
-			continue;
-		}
-		if( LineCarry )
-		{
-			switch( *p )
-			{
-			case L' ':
-			case L'\t':
-			case L'\r':
-				*p = L' ';
-				continue;
-			case L'\n':
-				currentLine++;
-				LineCarry = FALSE;
-				*p = L' ';
-				continue;
-			}
       *p = 0; // I2525
       wcscpy_s(wstr, LINESIZE, str);  // I3481
-			return (PreProcess ? CERR_None : CERR_InvalidLineContinuation);
-		}
-
-		if( *p == L'\n' ) break;
-		switch( *p )
-		{
-		case L'c':
-		case L'C':
-			if((p == str || iswspace(*(p-1))) && iswspace(*(p+1)) ) {
-				InComment = TRUE;
-				*p = L' ';
-				}
-			continue;
-		case L'\r':
-		case L'\t':
-			*p = L' ';
-			continue;
-		case L'\'':
-		case L'\"':
-			currentQuotes = *p;
-			continue;
-		}
-	}
-
-	if( n == len )
-  {
-    str[LINESIZE-1] = 0;  // I2525
-    wcscpy_s(wstr, LINESIZE, str);  // I3481
-		if(len == LINESIZE)
-	  	return (PreProcess ? CERR_None : CERR_LineTooLong);
+      return (PreProcess ? CERR_None : CERR_InvalidLineContinuation);
+    }
+
+    if (*p == L'\n') break;
+    switch (*p)
+    {
+    case L'c':
+    case L'C':
+      if ((p == str || iswspace(*(p - 1))) && iswspace(*(p + 1))) {
+        InComment = TRUE;
+        *p = L' ';
+      }
+      continue;
+    case L'\r':
+    case L'\t':
+      *p = L' ';
+      continue;
+    case L'\'':
+    case L'\"':
+      currentQuotes = *p;
+      continue;
     }
+  }
+
+  if (n == len)
+  {
+    str[LINESIZE - 1] = 0;  // I2525
+    wcscpy_s(wstr, LINESIZE, str);  // I3481
+    if (len == LINESIZE)
+      return (PreProcess ? CERR_None : CERR_LineTooLong);
+  }
 
-	if( *p == L'\n' ) currentLine++;
+  if (*p == L'\n') currentLine++;
 
-	SetFilePointer(hInfile, -(int)(len*2 - (INT_PTR)(p-str)*2 - 2), NULL, FILE_CURRENT);
+  SetFilePointer(hInfile, -(int)(len * 2 - (INT_PTR)(p - str) * 2 - 2), NULL, FILE_CURRENT);
 
-	p--;
-	while(p >= str && iswspace(*p)) p--;
-	p++;
-	*p++ = L'\n';
-	*p = 0;
-	// trim spaces now, why not?
+  p--;
+  while (p >= str && iswspace(*p)) p--;
+  p++;
+  *p++ = L'\n';
+  *p = 0;
+  // trim spaces now, why not?
   wcscpy_s(wstr, LINESIZE, str);  // I3481
 
-	return CERR_None;
+  return CERR_None;
 }
 
 DWORD GetRHS(PFILE_KEYBOARD fk, PWSTR p, PWSTR buf, int bufsize, int offset, int IsUnicode)
 {
-	PWSTR q;
+  PWSTR q;
 
-	p = wcschr(p, '>');
+  p = wcschr(p, '>');
 
-	if(!p) return CERR_NoTokensFound;
+  if (!p) return CERR_NoTokensFound;
 
-	p++;
+  p++;
 
-	return GetXString(fk, p, L"c\n", buf, bufsize, offset, &q, TRUE, IsUnicode);
+  return GetXString(fk, p, L"c\n", buf, bufsize, offset, &q, TRUE, IsUnicode);
 }
 
 void safe_wcsncpy(PWSTR out, PWSTR in, int cbMax)
 {
-	wcsncpy_s(out, cbMax, in, cbMax);  // I3481
-	out[cbMax-1] = 0;
+  wcsncpy_s(out, cbMax, in, cbMax - 1);  // I3481
```

Comment:
The `- 1` is new, I see.  What sort of out-of-bounds error does this address?  

I see similar edits to the same function (`wcsncpy_s`) and parameter in a few places later on too, so I'll assume it's the same kind of error there as well.

Reply:
The last parameter in `wcsncpy_s` is number of characters to copy, excluding nul terminator, so this was a buffer overflow, technically.